### PR TITLE
CDI: Fix deserialization of HttpSession bean

### LINKED ISSUE #29432: OpenLiberty Database Session Replication - org.jboss.weld.module.web.HttpSessionBean$SerializableProxy - ClassNotFoundException
I'm using OpenLiberty 24.0.0.6 configured with session replication. The application uses JSF with PrimeFaces. Some of my business objects are store in the HTTP session (via. @SessionScoped annotation of the CDI package). 

When the session is deserialized I get the following error (taken from the ffdc log): 

`
Exception = java.lang.ClassNotFoundException
Source = com.ibm.ws.session.SessionLoader.loadObject
probeid = 82
Stack Dump = java.lang.ClassNotFoundException: org.jboss.weld.module.web.HttpSessionBean$SerializableProxy
	at com.ibm.ws.classloading.internal.UnifiedClassLoader.findClass(UnifiedClassLoader.java:167)
	at com.ibm.ws.classloading.internal.ThreadContextClassLoader.findClass(ThreadContextClassLoader.java:145)
	at com.ibm.ws.classloading.internal.UnifiedClassLoader.loadClass0(UnifiedClassLoader.java:141)
	at com.ibm.ws.classloading.internal.UnifiedClassLoader$Delegation.loadClass(UnifiedClassLoader.java:87)
	at com.ibm.ws.classloading.internal.UnifiedClassLoader.loadClass(UnifiedClassLoader.java:106)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:527)
	at com.ibm.ws.serialization.DeserializationObjectInputStream.loadClass(DeserializationObjectInputStream.java:64)
	at com.ibm.ws.serialization.internal.DeserializationObjectInputStreamImpl.loadClass(DeserializationObjectInputStreamImpl.java:50)
	at com.ibm.ws.serialization.DeserializationObjectInputStream.resolveClass(DeserializationObjectInputStream.java:135)
	at com.ibm.ws.serialization.DeserializationObjectInputStream.resolveClass(DeserializationObjectInputStream.java:173)
	at java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2003)
	at java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1870)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2201)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1687)
	at java.base/java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2496)
	at java.base/java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2390)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2228)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1687)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:489)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:447)
	at java.base/java.util.Hashtable.readHashtable(Hashtable.java:1315)
	at java.base/java.util.Hashtable.readObject(Hashtable.java:1259)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.base/java.io.ObjectStreamClass.invokeReadObject(ObjectStreamClass.java:1046)
	at java.base/java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2357)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2228)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1687)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:489)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:447)
	at com.ibm.ws.session.utils.SessionLoader.loadObject(SessionLoader.java:64)
	at com.ibm.ws.session.store.db.DatabaseHashMap.getValue(DatabaseHashMap.java:1615)
`

The server is responding:

`
[15.08.24, 16:03:13:605 MESZ] 0000007c com.ibm.ws.session.WASSession                                E SESN0051E: Attempting to deserialize a session object from the backend resulted in a ClassNotFoundException..
[15.08.24, 16:03:13:605 MESZ] 0000007c com.ibm.ws.session.WASSession                                E ID: n3KSj82AVIMrmIK9ZH8A0WH
[15.08.24, 16:03:13:605 MESZ] 0000007c com.ibm.ws.session.WASSession                                E Exception: 
java.lang.ClassNotFoundException: org.jboss.weld.module.web.HttpSessionBean$SerializableProxy
`

I logged out the content of the HTTP session and I was wondering that a WELD proxy is part of the session: 

`
attribute name: 
WELD_S#F_WELD%AbstractBuiltInBean%web_war_exploded#web%HttpSession

attribute value (from toString())
Bean: ForwardingBean null for WELD%AbstractBuiltInBean%web_war_exploded#web%HttpSession; Instance: org.jboss.weld.module.web.HttpSessionBean$SerializableProxy@1bbaee59; CreationalContext: org.jboss.weld.contexts.CreationalContextImpl@27396b8a
`

I have no idea, why WELD is storing such a proxy class in the session. Btw. some content is stored in the database.

All the objects stored in the HTTP session are serializable. If not I think I would get an NotSerializableExcpeption (which is not in the logs or traces)

I was looking for the serialized data in the database and found the proxy class.

There are no WELD dependencies in the application (meaning the application war). The classpath look ok to me.

When I remove all injects of the HTTP session via CDI (@Inject), the error disappears. 

**Diagnostic information:**  
 - OpenLiberty Version: 24.0.0.7
 - Affected feature(s) javaee-8.0, microProfile-3.3 , monitor-1.0 , sessionDatabase-1.0
 - Java Version: OpenJDK 64-Bit Server VM, Version 11.0.23+9 (de_DE)
 - server.xml configuration (WITHOUT sensitive information like passwords)

`
<?xml version="1.0" encoding="UTF-8"?>
<server>
    
    <library id="SessionDriverLib">
        <fileset dir="${shared.resource.dir}/session" includes="*.jar"/>
    </library>

    <dataSource id="SessionDS">
        <jdbcDriver libraryRef="SessionDriverLib" />
        <properties.postgresql serverName="${postgres_server}"
                               portNumber="5432"
                               databaseName="httpSession"
                               user="session_user"
                               password="***"/>
    </dataSource>

    <httpSessionDatabase
            id="SessionDB"
            dataSourceRef="SessionDS"
            tableName="sessions"/>
    <httpSession storageRef="SessionDB" cloneId="${clone_id}"/>

</server>
`

**Additional context**  
See also the stackoverflow entry 
https://stackoverflow.com/questions/78878390/openliberty-database-session-replication-org-jboss-weld-module-web-httpsession

Thanks to Azquelt for the hint.

### PR DESCRIPTION
**Pull request by Azquelt**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

Fixes #29432

### CODE REVIEW

**Comment by Azquelt** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.cdi.2.0_fat,com.ibm.ws.cdi.annotations_fat,com.ibm.ws.cdi.api_fat,com.ibm.ws.cdi.beansxml_fat,com.ibm.ws.cdi.beansxml.implicit_fat,com.ibm.ws.cdi.ejb_fat,com.ibm.ws.cdi.extension_fat,com.ibm.ws.cdi.jee_fat,com.ibm.ws.cdi.jndi_fat,com.ibm.ws.cdi.lifecycle_fat,com.ibm.ws.cdi.noncontextual_fat,com.ibm.ws.cdi.third.party_fat,com.ibm.ws.cdi.visibility_fat,com.ibm.ws.clientcontainer.cdi_fat,com.ibm.ws.ejbcontainer.cdi_fat,com.ibm.ws.jaxrs.2.0.cdi.1.2_fat,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxws.cdi_fat,com.ibm.ws.jbatch.cdi_fat,com.ibm.ws.jpa.tests.spec21.cdi_fat,com.ibm.ws.microprofile.faulttolerance.cdi_fat,io.openliberty.cdi.3.0.internal_fat,io.openliberty.cdi.4.0.internal_fat,io.openliberty.cdi.4.1.internal_fat,io.openliberty.cdi.core_fat,io.openliberty.restfulWS.3.0.cdi.3.0_fat,com.ibm.ws.cdi.1.0_fat,com.ibm.ws.cdi.1.0_fat_EE,com.ibm.ws.cdi.1.0_fat_one,com.ibm.ws.cdi.1.0_fat_two,com.ibm.ws.cdi.1.0.mdb_fat,com.ibm.ws.cdi.1.2.mdb_fat,com.ibm.ws.ejbcontainer.cdi_ee6_fat

File: dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/session/CDISessionPersistenceTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/session/CDISessionPersistenceTest.java]: 2024

File: dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/FATSuite.java [hunk 2]
```diff
@@ -20,6 +20,7 @@
 import com.ibm.ws.cdi.jee.jsf.SimpleJSFWithSharedLibTest;
 import com.ibm.ws.cdi.jee.jsp.SimpleJSPTest;
 import com.ibm.ws.cdi.jee.servlet.ServletStartupTest;
+import com.ibm.ws.cdi.jee.session.CDISessionPersistenceTest;
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/FATSuite.java:23]: date

Comment (Azquelt) [reply dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/FATSuite.java:23]: Fixed

File: dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/session/CDISessionPersistenceTest.java (hunk 1 above)
Comment (Azquelt) [reply dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/session/CDISessionPersistenceTest.java]: Fixed

**Review by tevans78 — DISMISSED** [review]
LGTM

**Comment by Azquelt** [comment]
Tests show an error when a proxy for `ServletContext` is reflected upon, which can happen when it's serialized to persist the session.

I've raised #29447 for that issue and will remove `ServletContext` from this test until it's fixed.

**Comment by Azquelt** [comment]
#build

**Review by tevans78 — APPROVED** [review]

**Comment by Azquelt** [comment]
#libby

**Pull request merged**